packages/lodestar/src/sync/utils/sync.ts

```diff
@@ -0,0 +1,83 @@
+import {IReputation, ReputationStore} from "../IReputation";
+import {BeaconBlock, BeaconBlockHeader, Checkpoint, Epoch, Slot} from "@chainsafe/eth2.0-types";
+import {IBeaconConfig} from "@chainsafe/eth2.0-config";
+import {hashTreeRoot} from "@chainsafe/ssz";
+import {blockToHeader} from "../../chain/stateTransition/util";
+import {IReqResp} from "../../network";
+
+export function isValidHeaderChain(config: IBeaconConfig, start: BeaconBlockHeader, blocks: BeaconBlock[]): boolean {
+  blocks = blocks.sort((b1, b2) => b1.slot - b2.slot);
```

Comment:
feels sort of strange that this function mutates `blocks` (is relied upon in the calling code)